sphereon-kmp-crypto-kms-azure/src/commonMain/kotlin/com/sphereon/crypto/kms/azure/AzureKeyvaultClientConfig.kt

```diff
@@ -0,0 +1,90 @@
+package com.sphereon.crypto.kms.azure
+
+@kotlinx.serialization.Serializable
+data class AzureKeyvaultClientConfig(
+    val keyvaultUrl: String,
+    val tenantId: String,
+    val credentialOpts: CredentialOpts,
+    val hsmType: HSMType,
+    val applicationId: String? = null,
+    val headers: List<Header>? = null,
+    val exponentialBackoffRetryOpts: ExponentialBackoffRetryOpts? = null,
+)
+
+@kotlinx.serialization.Serializable
+data class CredentialOpts(
+    val credentialMode: CredentialMode,
+    val secretCredentialOpts: SecretCredentialOpts? = null,
+    val certificateCredentialOpts: CertificateCredentialOpts? = null,
+    val interactiveBrowserCredentialOpts: InteractiveBrowserCredentialOpts? = null,
+    val usernamePasswordCredentialOpts: UsernamePasswordCredentialOpts? = null
+)
+
+enum class CredentialMode(val credentialType: CredentialType) {
+    SERVICE_CLIENT_SECRET(CredentialType.SERVICE),
+    SERVICE_CLIENT_CERTIFICATE(CredentialType.SERVICE),
+    USER_INTERACTIVE_BROWSER(CredentialType.USER),
+    USER_USERNAME_PASSWORD(CredentialType.USER)
+}
+
+enum class CredentialType {
+    SERVICE, USER
+}
+
+enum class HSMType {
```

Comment:
@Serializable. Basically for all data classes and enums, as we will want to create a REST implementation at one point as well